python/taichi/misc/gui.py

```diff
@@ -75,6 +75,10 @@ def cook_image(self, img):
         assert res == self.res, "Image resolution does not match GUI resolution"
         return np.ascontiguousarray(img)
 
+    def get_image(self):
+        self.core.get_img(self.img.ctypes.data)
```

Comment:
What if self.img is not a contigugous array? We may want to enforce this in set_image.

Reply:
Agree, WIP

Reply:
Line 115:
```py
            self.img = self.cook_image(img)
```
Could you check if `cook_image` return a RGBA array? I didn't remember..

Reply:
Just now I double check it. If the input of `cook_image` is a valid image (no matter graylike or rgb or rgba), the return will be an RGBA array.

The only exception is when `img`'shape is [x, y, 2], in this case our `cook_image` will fail to return RGBA array. But [x, y, 2] is an invalid image size, IMO.